sdv/constraints/tabular.py

```diff
@@ -292,22 +325,31 @@ def reverse_transform(self, table_data):
         """
         table_data = table_data.copy()
         diff = (np.exp(table_data[self._diff_column]).round() - 1).clip(0)
-        if self._diff_is_datetime(table_data):
+        if self._is_datetime:
             diff = pd.to_timedelta(diff)
 
         if self._drop == 'high':
-            low_column = table_data[self._low]
-            table_data[self._high] = (low_column + diff).astype(self._dtype)
+            low = self._low if self._low_is_scalar else table_data[self._low]
+            table_data[self._high] = (low + diff).astype(self._dtype)
 
         elif self._drop == 'low':
-            high_column = table_data[self._high]
-            table_data[self._low] = (high_column - diff).astype(self._dtype)
+            high = self._high if self._high_is_scalar else table_data[self._high]
+            table_data[self._low] = (high - diff).astype(self._dtype)
 
         else:
-            low_column = table_data[self._low]
             invalid = ~self.is_valid(table_data)
-            new_high_values = low_column.loc[invalid] + diff.loc[invalid]
-            table_data[self._high].loc[invalid] = new_high_values.astype(self._dtype)
+            if self._high_is_scalar and not self._low_is_scalar:
```

Comment:
I think this if/else block could be simplified by only assigning `new_values` and then doing:

```python3
if ...
    new_values = ...
    column = ...
elif ...
    ...
else:
    ...

table_data[column].loc[invalid] = new_values.astype(self._dtype)
```

`column` should also be computed beforehand, in the `__init__`, since it will never change.

Reply:
I kind of don't like computing the column beforehand because we still have to do all the if/else statements here to calculate `new_values`, so we'd be doing that again earlier on. Also, it's decoupling the calculation of which column to use from the calculation of the values to replace. I think that could get confusing